samza-core/src/main/java/org/apache/samza/config/TaskConfigJava.java

```diff
@@ -37,7 +37,7 @@
 public class TaskConfigJava extends MapConfig {
   // Task Configs
   private static final String TASK_SHUTDOWN_MS = "task.shutdown.ms";
-  public static final long DEFAULT_TASK_SHUTDOWN_MS = 5000L;
+  public static final long DEFAULT_TASK_SHUTDOWN_MS = 30000L;
```

Comment:
> This change in worst case will cause long rebalancing phases 

This is definitely how this change will affect the Zk based deployment mode. However, it will affect the existing yarn based deployments. The side-effect may not be significant. But for high throughput jobs, a container downtime for 30 s means no processing for 30s that could cause significant lag. 
I am not in favor of changing existing default config values (http://samza.apache.org/learn/documentation/latest/jobs/configuration-table.html) as part of a larger patch. This kind of change should be treated equivalent to an api change and needs to be evaluated/planned separately. Can we please do that in a separate PR ?